### PR TITLE
MDL-72509 core: Move participants filter from core_user to core

### PR DESCRIPTION
**Pull request by TomoTsuyuki**
#310 De-couple the participants filter

This is just rebased and cleaned commits from old version => #321 

*** PLEASE DO NOT OPEN PULL REQUESTS VIA GITHUB ***

The moodle.git repository at Github is just a mirror of the official repository. We do not accept pull requests at Github.

See CONTRIBUTING.txt guidelines for how to contribute patches for Moodle. Thank you.

--

### CODE REVIEW

File: lib/amd/src/filter.js [hunk 1]
```diff
@@ -32,72 +31,130 @@ import Templates from 'core/templates';
...
+
+        this.filterSet = filterSet;
+        this.applyCallback = applyCallback;
+        // Keep a reference to all of the active filters.
+        this.activeFilters = {
+            courseid: new CourseFilter('courseid', filterSet),
+        };
+    }
+
+    init() {
```
Comment (mattporritt) [inline lib/amd/src/filter.js:52]: This will need some method docs

**Review by mattporritt — APPROVED** [review]
This looks good to continue with

File: lib/amd/src/filter.js (hunk 1 above)
Comment (TomoTsuyuki) [reply lib/amd/src/filter.js:55]: I added a comment to this function.

**Comment by safatshahin** [comment]
closing this as the branches are not valid anymore and already moved to tracker.

**Pull request closed without merging**